lib/src/component_declaration/annotations.dart

```diff
@@ -244,13 +273,16 @@ class PropsMixin implements TypedMap {
 /// State properties are declared as fields, which act as stubs for generated getters/setters that proxy Map key-value pairs.
 ///
 ///     @StateMixin()
-///     abstract class WaldoStateMixin {
+///     mixin WaldoStateMixin on UiState {
 ///       Map get state;
 ///
 ///       dynamic fred;
 ///     }
 ///
 /// Classes using this annotation must include the abstract `state` getter.
+///
+/// __NOTE:__ This is only required for legacy boilerplate and can be omitted
+/// for new implementations. However, it can still be used for custom configurations.
 class StateMixin implements TypedMap {
```

Comment:
From offline discussion: Let's deprecate `@StateMixin()` since they can just use `@State()` if they need to use something like the `keyNamespace` argument.

```suggestion
///
/// __Deprecated.__ Use the `@State()` annotation instead if you need to make use of an annotation argument. 
/// Otherwise, this can be removed completely. Will be removed in the 4.0.0 release of over_react.
@Deprecated('Use the @State() annotation if you need to make use of an annotation argument. Otherwise, this can be removed completely. Will be removed in the 4.0.0 release of over_react.')
class StateMixin implements TypedMap {
```